OAuth/ResourceOwner/AbstractResourceOwner.php

```diff
@@ -301,9 +301,9 @@ protected function generateNonce()
     /**
      * Configure the option resolver
      *
-     * @param OptionsResolverInterface $resolver
+     * @param OptionsResolver $resolver
      */
-    protected function configureOptions(OptionsResolverInterface $resolver)
+    protected function configureOptions(OptionsResolver $resolver)
```

Comment:
I know. This is why I changed it on each `ResourceOwner`.

This will affect resource owner adding PR that will be just have to be rebased with parameter change and custom made `ResourceOwner` classes.

Two solutions: 

 * Keep the legacy signature for 0.4, deprecated it and build a new one (`setUpOtions`?) Like Symfony did. Remove BC for next release (0.5 or 1.0).
 * Accept it as a BC as far 0.4 is not stable yet.

I'm not able to decide about it. Can you give me your opinion @stloyd?

**In any case**, we have indeed to add an upgrade note for it.